migrator/migrations/m_192_to_m_193_report_config_migrations/migration_impl.go

```diff
@@ -0,0 +1,221 @@
+	}
+	if rows.Err() != nil {
+		utils.Should(rows.Err())
+		return errors.Wrapf(rows.Err(), "failed to get rows for %s", "report_configurations")
+	}
+	defer func() { _ = rows.Close() }()
+	var convertedReportConfigs []*updatedSchema.ReportConfigurations
+	var reportSnapshots []*updatedSchema.ReportSnapshots
+	var reportNotifiers []*updatedSchema.ReportConfigurationsNotifiers
+	for rows.Next() {
+		var reportConfig *updatedSchema.ReportConfigurations
+		if err = db.ScanRows(rows, &reportConfig); err != nil {
+			return errors.Wrap(err, "failed to scan rows")
+		}
+		// convert to report config proto
+		reportConfigProto, err := updatedSchema.ConvertReportConfigurationToProto(reportConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to convert %+v to proto", reportConfigProto)
+		}
+		// skip if version=1 since config is migrated
+		// skip if version=2
+		if reportConfigProto.Version != 0 {
+			continue
+		}
+		if reportConfigProto.Version == 0 && reportConfigProto.GetResourceScope() != nil {
+			reportConfigProto.Version = 2
+			// convert report config proto back to gorm model
+			convertedGormConfig, err := updatedSchema.ConvertReportConfigurationFromProto(reportConfigProto)
+			if err != nil {
+				return errors.Wrapf(err, "failed to convert from proto %+v", reportConfigProto)
+			}
+			convertedReportConfigs = append(convertedReportConfigs, convertedGormConfig)
+
+			continue
+		}
+		// create v2 report config from v1
+		newConfig := create_v2_report_config(reportConfigProto)
+
+		// create notifier config for report_configuration_notifier table
+		notifierConfig := create_notifier(reportConfigProto)
+		newConfig.Notifiers = append(newConfig.Notifiers, notifierConfig)
+
+		// add notifier to notifier_configurations_notifiers
+		reportNotifierGorm, err := updatedSchema.ConvertNotifierConfigurationFromProto(notifierConfig, 0, newConfig.GetId())
+		if err != nil {
+			return errors.Wrapf(err, "failed to convert from proto %+v", notifierConfig)
+		}
+		reportNotifiers = append(reportNotifiers, reportNotifierGorm)
+
+		// assign version to 2 to new copy and version to 1 in original so that they are not re-created during migration
+		newConfig.Version = 2
+		reportConfigProto.Version = 1
+		// convert report config proto back to gorm model
+		convertedGormNewConfig, err := updatedSchema.ConvertReportConfigurationFromProto(newConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to convert from proto %+v", newConfig)
+		}
+		convertedGormReportConfig, err := updatedSchema.ConvertReportConfigurationFromProto(reportConfigProto)
+		if err != nil {
+			return errors.Wrapf(err, "failed to convert from proto %+v", reportConfigProto)
+		}
+		convertedReportConfigs = append(convertedReportConfigs, convertedGormNewConfig, convertedGormReportConfig)
+
+		// create report snapshot for last run report job
+		reportSnapshot := create_report_snapshot(reportConfigProto, newConfig)
+		if reportConfigProto.GetLastRunStatus() != nil {
```

Comment:
After you have made the suggested changes to `create_report_snapshpt` func, the check here would be
`if reportSnapshot != nil`